typescript-eslint recommended rules don't apply to gts files

## Repro
```sh
$ pnpm dlx ember-cli@latest addon my-addon -b @ember/addon-blueprint --pnpm --typescript
$ cd my-addon
$ echo 'export default function(assert: Assert) { return assert; }' > src/foo.ts
$ echo 'export default function(assert: Assert) { return assert; }' > src/bar.gts
$ pnpm lint:js
```

## Expected
Linting passes

## Actual
It passes for the `.ts` file, but not the `.gts` file:
```
> my-addon@0.0.0 lint:js /private/tmp/my-addon
> eslint . --cache


/private/tmp/my-addon/src/bar.gts
  1:33  error  'Assert' is not defined  no-undef

✖ 1 problem (1 error, 0 warnings)

 ELIFECYCLE  Command failed with exit code 1.
```

## Discussion
The problem is [here](https://github.com/ember-cli/ember-addon-blueprint/blob/81e8897b22da834d90f5e002961dc57e6a720395/files/eslint.config.mjs#L65-L75). `ts.configs.recommendedTypeChecked` [specifies a files array](https://github.com/typescript-eslint/typescript-eslint/blob/2929b876042dca10b9d2c3cbedbc6857a3a403f8/packages/eslint-plugin/src/configs/eslint-recommended-raw.ts#L14-L19), but as outlined in the [tseslint.config to defineConfig migration guide](https://typescript-eslint.io/packages/typescript-eslint/#migrating-to-defineconfig), overriding the files array (now) results in the intersection of the two arrays, which in this case ends up just being `ts`, so it doesn't apply to `gts` files and therefore, among other things, doesn't turn off `no-undef`.

I can't see a super clean way to address this directly because I don't think `typescript-eslint` exports just that recommended ruleset without the files specified. Perhaps we should PR the addition of `gts` to that files array (and any other relevant files arrays in `typescript-eslint`)? In the near term, we could work around it by switching back to the deprecated `tseslint.config()`, or with some non-ideal hackery:

```js
    {
      files: ['**/*.{ts,gts}'],
      // *snip*
      extends: [
        ...ts.configs.recommendedTypeChecked.map((entry) => {
          const processed = { ...entry };
          delete processed.files;
          return processed;
        }),
        ember.configs.gts,
      ],
    },
```

I'm happy to PR a change into `typescript-eslint` but perhaps somebody else in the community already has a relationship and would be more likely to have the PR accepted if they were the author? Also happy to open a PR into this repo for one of the workarounds with some guidance on which way we think we want to go.

NOTE: this also affects the app blueprint.

--- a/files/eslint.config.mjs
+++ b/files/eslint.config.mjs
@@ -71,7 +71,15 @@ export default defineConfig([
         ...globals.browser,
       },
     },
-    extends: [...ts.configs.recommendedTypeChecked, ember.configs.gts],
+    extends: [
+      ...ts.configs.recommendedTypeChecked,
+      // https://github.com/ember-cli/ember-addon-blueprint/issues/119
+      {
+        ...ts.configs.eslintRecommended,
+        files: undefined,
+      },
+      ember.configs.gts,
+    ],
   },<% } %>
   {
     files: ['src/**/*'],

--- a/tests/fixtures/typescript/src/lint-test-gts.gts
+++ b/tests/fixtures/typescript/src/lint-test-gts.gts
@@ -1,0 +1,15 @@
+/**
+ * This file is used to ensure that `eslint.config.mjs` is properly configured
+ * to apply `typescript-eslint`'s recommended rules to `.gts` files. It ensures
+ * that:
+ *
+ * - `no-undef` is disabled (otherwise the undefined symbol would cause a
+ *   linting error)
+ * - `@typescript-eslint/no-unsafe-return` is enabled (otherwise the
+ *   `eslint-disable-next-line` comment would cause a linting error)
+ */
+export default function () {
+  // @ts-expect-error testing lint
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-return
+  return notDefined;
+}

--- a/tests/fixtures/typescript/src/lint-test-ts.ts
+++ b/tests/fixtures/typescript/src/lint-test-ts.ts
@@ -1,0 +1,15 @@
+/**
+ * This file is used to ensure that `eslint.config.mjs` is properly configured
+ * to apply `typescript-eslint`'s recommended rules to `.ts` files. It ensures
+ * that:
+ *
+ * - `no-undef` is disabled (otherwise the undefined symbol would cause a
+ *   linting error)
+ * - `@typescript-eslint/no-unsafe-return` is enabled (otherwise the
+ *   `eslint-disable-next-line` comment would cause a linting error)
+ */
+export default function () {
+  // @ts-expect-error testing lint
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-return
+  return notDefined;
+}

--- a/tests/smoke-tests/--typescript.test.ts
+++ b/tests/smoke-tests/--typescript.test.ts
@@ -97,6 +97,8 @@ for (let packageManager of SUPPORTED_PACKAGE_MANAGERS) {
         ).toMatchInlineSnapshot(`
           [
             "src/index.ts",
+            "src/lint-test-gts.gts",
+            "src/lint-test-ts.ts",
             "src/template-registry.ts",
             "src/components/another-gts.gts",
             "src/components/template-import.gts",
@@ -111,6 +113,10 @@ for (let packageManager of SUPPORTED_PACKAGE_MANAGERS) {
           [
             "dist/index.js",
             "dist/index.js.map",
+            "dist/lint-test-gts.js",
+            "dist/lint-test-gts.js.map",
+            "dist/lint-test-ts.js",
+            "dist/lint-test-ts.js.map",
             "dist/template-registry.js",
             "dist/template-registry.js.map",
             "dist/components/another-gts.js",
@@ -124,6 +130,10 @@ for (let packageManager of SUPPORTED_PACKAGE_MANAGERS) {
             "dist/_app_/services/example.js",
             "declarations/index.d.ts",
             "declarations/index.d.ts.map",
+            "declarations/lint-test-gts.d.ts",
+            "declarations/lint-test-gts.d.ts.map",
+            "declarations/lint-test-ts.d.ts",
+            "declarations/lint-test-ts.d.ts.map",
             "declarations/template-registry.d.ts",
             "declarations/template-registry.d.ts.map",
             "declarations/components/another-gts.d.ts",

